src/internal.c

```diff
@@ -3301,31 +3301,29 @@ static int ParseAndVerifyCert(WOLFSSH* ssh, byte* in, word32 inSz,
     int ret;
     word32 l = 0, m = 0;
     word32 ocspCount = 0;
-    byte*  ocspBuf   = NULL;
-    word32 ocspBufSz = 0;
     word32 certCount = 0;
-    byte*  certPt    = NULL;
+    byte*  certChain = NULL;
     word32 certChainSz = 0;
 
     /* Skip the name */
     ret = GetSize(&l, in, inSz, &m);
     m += l;
 
-    /* Get the cert count */
-    ret = GetUint32(&certCount, in, inSz, &m);
     if (ret == WS_SUCCESS) {
-        WLOG(WS_LOG_INFO, "Peer sent certificate count of %d", certCount);
+        /* Get the cert count */
+        ret = GetUint32(&certCount, in, inSz, &m);
     }
 
     if (ret == WS_SUCCESS) {
         word32 count;
 
-        certPt = in + m;
-        m = 0;
+        WLOG(WS_LOG_INFO, "Peer sent certificate count of %d", certCount);
+        certChain = in + m;
+
         for (count = certCount; count > 0; count--) {
             word32 certSz = 0;
 
-            ret = GetSize(&certSz, certPt, inSz, &m);
+            ret = GetSize(&certSz, in, inSz, &m);
             WLOG(WS_LOG_INFO, "Adding certificate size %d", certSz);
```

Comment:
Same issue with %d here. Would also prefer this WLOG get moved after the break on ret != WS_SUCCESS block.

Reply:
got it